lib/components/viewers/pattern-row.tsx

```diff
@@ -68,113 +72,63 @@ class PatternRow extends Component<Props, State> {
     }
 
     const routeName = route.shortName ? route.shortName : route.longName
+    const routeColor = getRouteColorBasedOnSettings(route.operator, route)
 
     return (
-      <div className="route-row" role="table">
+      <div className="route-row" role="group">
         {/* header row */}
-        <div className="header" role="row">
+        <div
+          className="header"
+          style={{
+            backgroundColor: routeColor,
+            color: getMostReadableTextColor(routeColor, route?.textColor)
+          }}
+        >
           {/* route name */}
           <div className="route-name">
-            <strong>
-              <RouteRenderer leg={generateFakeLegForRouteRenderer(route)} />
+            <strong
+              style={
+                routeName && routeName?.length >= 4 ? { fontSize: '3vb' } : {}
+              }
+            >
+              <div
```

Comment:
If we did this the flex box would affect the headsign as well. We could accomplish the div removal by using grid but that would add a lot of complexity to what is otherwise a simple component (style-wise).